packages/@divvi/mobile/src/divviProtocol/slice.ts

```diff
@@ -0,0 +1,49 @@
+import { createSlice, PayloadAction } from '@reduxjs/toolkit'
+import { getRehydratePayload, REHYDRATE, RehydrateAction } from 'src/redux/persist-helper'
+import { RootState } from 'src/redux/store'
+
+interface ReferralState {
+  successfulReferrals: {
+    [key: string]: boolean // key is `${divviId}-${campaignIds.join(',')}` where campaignIds are sorted
+  }
+}
+
+const initialState: ReferralState = {
+  successfulReferrals: {},
+}
+
+const divviProtocolSlice = createSlice({
+  name: 'divviProtocol',
+  initialState,
+  reducers: {
+    markReferralSuccessful: (
```

Comment:
nit: we should [model actions as events](https://redux.js.org/style-guide/#model-actions-as-events-not-setters).
```suggestion
    referralSubmissionSucceeded: (
```